### PR TITLE
 feat: add docker and docker compose compute tasks 

### PR DESCRIPTION
**Pull request by kshychko**
## Intent of Change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Description

This PR introduces two new extensions - docker and docker-compose.
## Motivation and Context

In order to support running docker-compose projects on ec2 instance docker and docker-compose need to be installed there during cfn init stage.
## How Has This Been Tested?

Tested locally
### Prerequisite PRs:

- None

### Dependent PRs:

- None

### Consumer Actions:

- None

### CODE REVIEW

File: aws/extensions/computetask_linux_docker_compose/extension.ftl [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            "ignoreErrors" : false
+        },
+        "02ApplyExecutablePermission" : {
+            "command" : "chmod +x /usr/local/bin/docker-compose",
+            "ignoreErrors" : false
+        },
+        "02ApplyExecutablePermission" : {
+            "command" : "chmod +x /usr/local/bin/docker-compose",
+            "ignoreErrors" : false
+        },
```
Comment (roleyfoley) [inline aws/extensions/computetask_linux_docker_compose/extension.ftl:37]: Looks like a duplicate of the same script

File: aws/extensions/computetask_linux_docker_compose/extension.ftl [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        "03AddLink" : {
+            "command" : "ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose",
+            "ignoreErrors" : false
+        },
```
Comment (roleyfoley) [inline aws/extensions/computetask_linux_docker_compose/extension.ftl:41]: is /usr/local/bin not on the path?

File: aws/extensions/computetask_linux_docker/extension.ftl [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        }
+    }]
+
+    [#local commands += {
+        "01AddUserToDockerGroup" : {
+            "command" : "usermod -a -G docker ec2-user",
+            "ignoreErrors" : false
+        }
+    }]
+
```
Comment (roleyfoley) [inline aws/extensions/computetask_linux_docker/extension.ftl:68]: Might be worth putting this under the OS switch statement. The ec2-user will only work on awslinux right?

**Review by roleyfoley — CHANGES_REQUESTED** [review]

File: aws/extensions/computetask_linux_docker/extension.ftl [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            context={ "OccurrenceId" : core.Id, "OperatingSystem" : operatingSystem }
+        /]
+    [/#if]
+
+
+    [#local commands += {
+        "01AddUserToDockerGroup" : {
+            "command" : "service docker start",
+            "ignoreErrors" : false
+        },
```
[the remark this answers is not in the document]
Comment (kshychko) [reply aws/extensions/computetask_linux_docker/extension.ftl]: @roleyfoley , seems that comes from an old commit before I force-pushed the branch
now it is in a service section.

File: aws/extensions/computetask_linux_docker/extension.ftl (hunk 3 above)
Comment (kshychko) [reply aws/extensions/computetask_linux_docker/extension.ftl]: moved

File: aws/extensions/computetask_linux_docker_compose/extension.ftl (hunk 2 above)
Comment (kshychko) [reply aws/extensions/computetask_linux_docker_compose/extension.ftl]: removed

File: aws/extensions/computetask_linux_docker_compose/extension.ftl (hunk 1 above)
Comment (kshychko) [reply aws/extensions/computetask_linux_docker_compose/extension.ftl]: removed

**Review by roleyfoley — APPROVED** [review]

**Pull request merged**